### PR TITLE
Adding Eventarc GoogleChannelConfig Resource support for TPG 

### PR DESCRIPTION
**Pull request by daveavi**
I have made changes to the magic-modules repo to support a new resource the Eventarc team has created, called GoogleChannelConfig. It is a regional singleton resource that enables customers to proctect their data with encryption for all first party triggers. 

Fixes: https://github.com/hashicorp/terraform-provider-google/issues/12520

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
google_eventarc_google_channel_config
```

### CODE REVIEW

**Comment by rileykarson** [comment]
note: holding review for https://github.com/GoogleCloudPlatform/magic-modules/pull/6529 to get merged (or ready for merge), since I'd imagine comments will be similar between them

File: tpgtools/api/eventarc/samples/copy.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import os
```
Comment (rileykarson) [inline tpgtools/api/eventarc/samples/copy.py:1]: Hmm- I'm not really sure what this is from / what it's for?

File: mmv1/third_party/terraform/tests/resource_eventarc_google_channel_config_test.go.erb [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+
+	dcl "github.com/GoogleCloudPlatform/declarative-resource-client-library/dcl"
+	<% if version == "ga" -%>
+	eventarc "github.com/GoogleCloudPlatform/declarative-resource-client-library/services/google/eventarc"
+	<% else -%>
+    eventarc "github.com/GoogleCloudPlatform/declarative-resource-client-library/services/google/eventarc/beta"
```
Comment (rileykarson) [inline mmv1/third_party/terraform/tests/resource_eventarc_google_channel_config_test.go.erb:14]: nit: These are spaces while other lines are tabs, would you mind converting these?

It's a template file, so editors don't work _quite_ right.

**Review by rileykarson — COMMENTED** [review]
Couple minor points- some formatting and a stray file- but otherwise good to merge once the DCL version's been updated in a separate PR and we've rebased or otherwise resolved the merge conflicts.

File: tpgtools/api/eventarc/samples/copy.py (hunk 1 above)
Comment (daveavi) [reply tpgtools/api/eventarc/samples/copy.py:1]: Ohh yes I need to remove this file, this was a script I ran to copy samples from DCL into this directory, will delete this

File: mmv1/third_party/terraform/tests/resource_eventarc_google_channel_config_test.go.erb (hunk 2 above)
Comment (daveavi) [reply mmv1/third_party/terraform/tests/resource_eventarc_google_channel_config_test.go.erb:14]: Acknowledged

Comment (daveavi) [reply mmv1/third_party/terraform/tests/resource_eventarc_google_channel_config_test.go.erb]: Done

File: tpgtools/api/eventarc/samples/copy.py (hunk 1 above)
Comment (daveavi) [reply tpgtools/api/eventarc/samples/copy.py]: Delete the file

**Review by rileykarson — COMMENTED** [review]
Two small comments, one inline and one here;

We'll need to add a GA copy of tpgtools/overrides/eventarc/beta/google_channel_config.yaml as well, this currently only adds a beta version of the resource.

**Review by rileykarson — APPROVED** [review]
LGTM, thanks!

**Comment by daveavi** [comment]
Thanks Riley!

**Pull request merged**